pkg/replicaset/replicaset.go

```diff
@@ -0,0 +1,446 @@
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// WithLabel applies label to replicaset's definition.
+func (builder *Builder) WithLabel(labelKey, labelValue string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(fmt.Sprintf("Defining replicaset's label to %s: %s", labelKey, labelValue))
+
+	if labelKey == "" {
+		glog.V(100).Infof("The 'labelKey' of the replicaset is empty")
+
+		builder.errorMsg = "can not apply empty labelKey"
+
+		return builder
+	}
+
+	if builder.Definition.Labels == nil {
+		builder.Definition.Labels = map[string]string{}
+	}
+
+	builder.Definition.Labels[labelKey] = labelValue
+
+	return builder
+}
+
+// WithNodeSelector applies nodeSelector to the replicaset definition.
+func (builder *Builder) WithNodeSelector(nodeSelectorKey, nodeSelectorValue string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying nodeSelector %s: %s to replicaset %s in namespace %s",
+		nodeSelectorKey, nodeSelectorValue, builder.Definition.Name, builder.Definition.Namespace)
+
+	if nodeSelectorKey == "" {
+		glog.V(100).Infof("The 'nodeSelectorKey' of the replicaset is empty")
+
+		builder.errorMsg = "can not apply empty nodeSelectorKey"
+
+		return builder
+	}
+
+	if builder.Definition.Spec.Template.Spec.NodeSelector == nil {
+		builder.Definition.Spec.Template.Spec.NodeSelector = map[string]string{}
+	}
+
+	builder.Definition.Spec.Template.Spec.NodeSelector[nodeSelectorKey] = nodeSelectorValue
+
+	return builder
+}
+
+// WithVolume defines Volume of replicaset under ContainerTemplateSpec.
+func (builder *Builder) WithVolume(dsVolume corev1.Volume) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if dsVolume.Name == "" {
+		glog.V(100).Infof("The Volume name parameter is empty")
+
+		builder.errorMsg = "Volume name parameter is empty"
+
+		return builder
+	}
+
+	glog.V(100).Infof("Adding volume %s for replicaset %s container template in namespace %s",
+		dsVolume.Name, builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.Volumes = append(
+		builder.Definition.Spec.Template.Spec.Volumes,
+		dsVolume)
+
+	return builder
+}
+
+// WithAdditionalContainerSpecs appends a list of container specs to the replicaset definition.
+func (builder *Builder) WithAdditionalContainerSpecs(specs []corev1.Container) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Appending a list of container specs %v to replicaset %s in namespace %s",
+		specs, builder.Definition.Name, builder.Definition.Namespace)
+
+	if len(specs) == 0 {
+		glog.V(100).Infof("The container specs are empty")
+
+		builder.errorMsg = "cannot accept empty list as container specs"
+
+		return builder
+	}
+
+	if builder.Definition.Spec.Template.Spec.Containers == nil {
```

Comment:
No need for this validation. 
Simply use: 
`builder.Definition.Spec.Template.Spec.Containers = append(builder.Definition.Spec.Template.Spec.Containers, specs...)`